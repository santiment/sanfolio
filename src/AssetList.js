import React from 'react'
import AssetItem from './AssetItem'

const AssetList = ({markets, prices, history}) => (
  <div className='assets-list'>
    {Object.keys(markets)
      .filter((key, index) => index < 20)
      .map((title, index) => (
        <AssetItem
          key={index}
          asset={{
            title,
            history: history && history[title] ? history[title] : null,
            price: prices[title],
            cap: markets[title]
          }} />
    ))}
  </div>
)

export default AssetList
