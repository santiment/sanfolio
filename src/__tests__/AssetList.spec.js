/* eslint-env jasmine */
import React from 'react'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import AssetList from '../AssetList'
configure({ adapter: new Adapter() })

describe('AssetList', () => {
  let wrapper

  const markets = {
    'BTC': 1202021,
    'ETH': 500000
  }

  const prices = {
    'BTC': 120,
    'ETH': 500
  }

  const history = {
    'BTC': [119],
    'ETH': [400]
  }

  beforeEach(() => {
    wrapper = shallow(<AssetList
      markets={markets}
      prices={prices}
      history={history} />)
  })

  it('should render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('render the DUMB component', () => {
    expect(wrapper.length).toEqual(1)
  })
})
