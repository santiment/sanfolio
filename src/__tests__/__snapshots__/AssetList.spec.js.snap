// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetList should render correctly 1`] = `
<div
  className="assets-list"
>
  <pure(withState(lifecycle(AssetItem)))
    asset={
      Object {
        "cap": 1202021,
        "history": Array [
          119,
        ],
        "price": 120,
        "title": "BTC",
      }
    }
    key="0"
  />
  <pure(withState(lifecycle(AssetItem)))
    asset={
      Object {
        "cap": 500000,
        "history": Array [
          400,
        ],
        "price": 500,
        "title": "ETH",
      }
    }
    key="1"
  />
</div>
`;
